--- brainworkshop/brainworkshop.pyw.orig	2013-03-28 08:55:44.654083679 +0700
+++ brainworkshop/brainworkshop.pyw	2013-03-28 09:05:31.465263050 +0700
@@ -35,8 +35,10 @@
 NOVBO = True
 VSYNC = False
 DEBUG = False
-FOLDER_RES = 'res'
-FOLDER_DATA = 'data'
+FOLDER_RES = '/usr/share/games/brainworkshop'
+FOLDER_DATA = os.path.expanduser('~/brainworkshop')
+if not os.path.isdir(FOLDER_DATA):
+	os.mkdir(FOLDER_DATA)
 CONFIGFILE = 'config.ini'
 STATS_BINARY = 'logfile.dat'
 USER = 'default'
