Last-Update: 2012-03-27
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Forwarded: yes
Bug-Debian: http://bugs.debian.org/665584
Description: fix FTBFS with glib 2.32


--- a/backend/wbprivate/sqlide/wb_live_schema_tree.h
+++ b/backend/wbprivate/sqlide/wb_live_schema_tree.h
@@ -24,7 +24,7 @@
 #include "grtpp.h"
 #include "workbench/wb_backend_public_interface.h"
 #include "base/string_utilities.h"
-#include <glib/gpattern.h>
+#include <glib.h>
 
 namespace wb
 {
--- a/backend/wbprivate/workbench/stdafx.h
+++ b/backend/wbprivate/workbench/stdafx.h
@@ -56,7 +56,6 @@
 #include <gl/glu.h>
 
 #include <glib.h>
-#include <glib/gthread.h>
 
 #include "grtpp.h"
 #include "grtpp_util.h"
--- a/library/canvas/src/mdc_canvas_view.h
+++ b/library/canvas/src/mdc_canvas_view.h
@@ -28,7 +28,7 @@
 #include "mdc_selection.h"
 
 #ifndef _WIN32
-#include <glib/gthread.h>
+#include <glib.h>
 #endif
 
 BEGIN_MDC_DECLS
--- a/library/canvas/src/mdc_canvas_view_printing.cpp
+++ b/library/canvas/src/mdc_canvas_view_printing.cpp
@@ -21,7 +21,7 @@
 
 #include "mdc_canvas_view.h"
 #include "mdc_canvas_view_printing.h"
-#include <glib/gmessages.h>
+#include <glib.h>
 
 #ifdef _WIN32
 #include <cairo/cairo-win32.h>
--- a/library/canvas/src/stdafx.h
+++ b/library/canvas/src/stdafx.h
@@ -52,7 +52,7 @@
 #include <gl/gl.h>
 #include <gl/glu.h>
 
-#include <glib/gthread.h>
+#include <glib.h>
 
 #endif // _WIN32
 
--- a/library/grt/src/grtpp_util.cpp
+++ b/library/grt/src/grtpp_util.cpp
@@ -25,7 +25,7 @@
 #include "grtpp_util.h"
 #include "base/string_utilities.h"
 
-#include <glib/gbase64.h>
+#include <glib.h>
 #include <functional>
 
 using namespace grt;
--- a/modules/db.mysql.sqlparser/src/mysql_sql_parser_fe.cpp
+++ b/modules/db.mysql.sqlparser/src/mysql_sql_parser_fe.cpp
@@ -20,7 +20,7 @@
 
 #include "stdafx.h"
 
-#include <glib/gunicode.h>
+#include <glib.h>
 #include "grt/common.h"
 #include "grt/grt_manager.h"
 #include "grtdb/charset_utils.h"
--- a/modules/db.mysql.sqlparser/src/stdafx.h
+++ b/modules/db.mysql.sqlparser/src/stdafx.h
@@ -32,7 +32,7 @@
 #include <algorithm>
 #include <sstream>
 #include <fstream>
-#include <glib/gunicode.h>
+#include <glib.h>
 #include <map>
 #include "direct.h"
 
--- a/modules/wb.mysql.import/src/stdafx.h
+++ b/modules/wb.mysql.import/src/stdafx.h
@@ -10,7 +10,7 @@
 #include <algorithm>
 #include <sstream>
 #include <fstream>
-#include <glib/gunicode.h>
+#include <glib.h>
 #include <map>
 #include "direct.h"
 
--- a/plugins/db.mysql/backend/db_plugin_be.cpp
+++ b/plugins/db.mysql/backend/db_plugin_be.cpp
@@ -31,7 +31,7 @@
 #include "grts/structs.db.h"
 #include "base/string_utilities.h"
 
-#include <glib/gunicode.h>
+#include <glib.h>
 
 void Db_plugin::grtm(bec::GRTManager *grtm)
 {
--- a/plugins/wb.printing/backend/stdafx.h
+++ b/plugins/wb.printing/backend/stdafx.h
@@ -44,7 +44,7 @@
 #include <gl/gl.h>
 #include <gl/glu.h>
 
-#include <glib/gthread.h>
+#include <glib.h>
 
 #ifndef _WIN32
 #include <sys/time.h>
