--- vmnet-only/vmnetInt.h.orig  2016-01-14 09:04:40.248877530 +0700
+++ vmnet-only/vmnetInt.h       2016-01-14 09:05:26.570307176 +0700
@@ -78,8 +78,19 @@
 
 extern struct proto vmnet_proto;
 #ifdef VMW_NETDEV_HAS_NET
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 2, 0)
+
 #   define compat_sk_alloc(_bri, _pri) sk_alloc(&init_net, \
+
                                                 PF_NETLINK, _pri, &vmnet_proto)
+
+#else
+
+#   define compat_sk_alloc(_bri, _pri) sk_alloc(&init_net, \
+
+                                                AF_NETLINK, _pri, &vmnet_proto, 1)
+
+#endif
 #else
 #   define compat_sk_alloc(_bri, _pri) sk_alloc(PF_NETLINK, _pri, &vmnet_proto, 1)
 #endif
