--- OpenBoard-1.6.0a/OpenBoard.pro.orig	2020-04-07 08:25:47.015291094 +0700
+++ OpenBoard-1.6.0a/OpenBoard.pro	2020-04-07 08:26:15.093547854 +0700
@@ -72,9 +72,7 @@
 INCLUDEPATH += plugins/cffadaptor/src
 
 #ThirdParty
-DEPENDPATH += $$THIRD_PARTY_PATH/quazip/
-INCLUDEPATH += $$THIRD_PARTY_PATH/quazip/
-include($$THIRD_PARTY_PATH/quazip/quazip.pri)
+INCLUDEPATH += /usr/include/quazip5/
 
 FORMS += resources/forms/mainWindow.ui \
    resources/forms/preferences.ui \
--- OpenBoard-1.6.0a/plugins/cffadaptor/UBCFFAdaptor.pro.orig	2020-04-07 08:26:29.240677360 +0700
+++ OpenBoard-1.6.0a/plugins/cffadaptor/UBCFFAdaptor.pro	2020-04-07 08:27:03.695993158 +0700
@@ -39,7 +39,7 @@
     QMAKE_POST_LINK += bash postScript_mac.sh "$$DESTDIR" "$$SANKORE_DIR/$$BUILD_DIR/product/Open-Sankore.app/Contents/MacOS"
 }
 
-LIBS     += "-L$$THIRD_PARTY_PATH/quazip/lib/$$SUB_DIR" "-lquazip"
+LIBS     += "-lquazip5"
 
 QT       += xml xmlpatterns core
 QT       += gui
@@ -51,10 +51,7 @@
 
 INCLUDEPATH += src
 
-DEPENDPATH  += $$THIRD_PARTY_PATH/quazip/
-INCLUDEPATH += $$THIRD_PARTY_PATH/quazip/
-include($$THIRD_PARTY_PATH/quazip/quazip.pri)
-
+INCLUDEPATH += /usr/include/quazip5/
 
 SOURCES += \
     src/UBCFFAdaptor.cpp
@@ -66,4 +63,4 @@
     src/UBCFFConstants.h
 
 RESOURCES += \
-    resources/resources.qrc
\ No newline at end of file
+    resources/resources.qrc
